artemis-commons/src/main/java/org/apache/activemq/artemis/api/core/SimpleString.java

```diff
@@ -454,7 +454,16 @@ public boolean contains(final char c) {
     * @return the concatenated SimpleString
     */
    public SimpleString concat(final String toAdd) {
-      return concat(new SimpleString(toAdd));
+      int len = toAdd.length();
```

Comment:
This concat method need test coverage IMO

Reply:
I assume as existing code its already covered

Reply:
It basically avoiding a new object and a byte array being created and thrown away uneededly. If its contentious i can revert, as not actually needed for this feature, just something i noticed when looking at object creation when profiling during dev of the feature.

Reply:
there is also explicit tests for concat in
org.apache.activemq.artemis.tests.util.SimpleStringTest.testConcat